### PR TITLE
UltraStaffChat - Version 5.2.0

### PR DESCRIPTION
**Pull request by LooFifteen**
Changes:
* Update to 1.18
* Fix bug where join server is unknown
* Update dependencies

### CODE REVIEW

File: src/main/java/dev/hypera/ultrastaffchat/events/staff/StaffJoinEvent.java [hunk 1]
```diff
@@ -29,11 +31,23 @@
 public class StaffJoinEvent extends PostLoginEvent implements Cancellable {
 
 	private boolean cancelled = false;
+	private Server server;
```
Comment (SLLCoding) [inline src/main/java/dev/hypera/ultrastaffchat/events/staff/StaffJoinEvent.java:34]: Make this final you nitwit

File: src/main/java/dev/hypera/ultrastaffchat/events/staff/StaffSwitchServerEvent.java [hunk 2]
```diff
@@ -18,35 +18,59 @@
...
 import net.md_5.bungee.api.plugin.Event;
+import org.jetbrains.annotations.ApiStatus;
 
 public class StaffSwitchServerEvent extends Event implements Cancellable {
 
 	private final ProxiedPlayer player;
 	private final String from;
 	private final String to;
+	private ServerInfo fromServer;
+	private Server toServer;
```
Comment (SLLCoding) [inline src/main/java/dev/hypera/ultrastaffchat/events/staff/StaffSwitchServerEvent.java:34]: again

**Review by SLLCoding — COMMENTED** [review]
c'mon me

**Comment by SLLCoding** [comment]
scrap that, I don't know how to read

**Comment by LooFifteen** [comment]
Needs to be tested and minor bumped.

**Comment by joshuasing** [comment]
^ rebase onto main to fix merge issues

**Comment by joshuasing** [comment]
It would be nice to have this merged as soon as possible

**Review by joshuasing — APPROVED** [review]

**Review by LooFifteen — COMMENTED** [review]
couple changes

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
[the remark this answers is not in the document]
Comment (joshuasing) [reply gradle/wrapper/gradle-wrapper.properties:22]: I could try, but I don't think Gradle 8.2 will work until Blossom updates

File: build-logic/src/main/kotlin/ultrastaffchat.publishing.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            developers {
+                developer {
+                    id.set("joshuasing")
+                    name.set("Joshua Sing")
+                    timezone.set("Australia/Melbourne")
+                    email.set("joshua@hypera.dev")
+                }
+
+                developer {
+                    id.set("SLLCoding")
```
[the remark this answers is not in the document]
Comment (joshuasing) [reply build-logic/src/main/kotlin/ultrastaffchat.publishing.gradle.kts:50]: Oops

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshuasing) [reply gradle/wrapper/gradle-wrapper.properties:22]: In a private conversation we've agreed to update to Gradle 8.2 after this release.

**Review by joshuasing — APPROVED** [review]

**Pull request merged**